crypto/data_formats.go

```diff
@@ -53,3 +55,66 @@ func HashLogRoot(root trillian.SignedLogRoot) ([]byte, error) {
 	}
 	return hash[:], nil
 }
+
+// LogRootV1 contains the fields verified by SignedLogRoot
+type LogRootV1 struct {
+	Version        uint16
+	TreeSize       uint64
+	RootHash       []byte `tls:"minlen:0,maxlen:128"`
+	TimestampNanos uint64
+	Revision       uint64
+	Metadata       []byte `tls:"minlen:0,maxlen:65535"`
+}
+
+// MapRootV1 contains the fields verified by SignedMapRoot
+type MapRootV1 struct {
+	Version        uint16
+	RootHash       []byte `tls:"minlen:0,maxlen:128"`
+	TimestampNanos uint64
+	Revision       uint64
+	Metadata       []byte `tls:"minlen:0,maxlen:65535"`
+}
+
+// ParseLogRoot returns a *SignedLogRootV1
+func ParseLogRoot(b []byte) (*LogRootV1, error) {
+	// Verify version
+	version := binary.BigEndian.Uint16(b)
+	if got, want := version, uint16(trillian.LogRootFormat_LOG_ROOT_FORMAT_V1); got != want {
```

Comment:
Personally, I'm not a big fan of using the `got, want:=...` thing outside of unit tests, unless there's a calculation to avoid repeating -- I think it's easier to see what's going on with:
```go
 if version != uint16(trillian.LogRootFormat_LOG_ROOT_FORMAT_V1) {
```
(But I know other opinions vary...)

Reply:
done